metricbeat/helper/sql/sql.go

```diff
@@ -44,22 +44,22 @@ type sqlRow interface {
 
 // NewDBClient gets a client ready to query the database
 func NewDBClient(driver, uri string, l *logp.Logger) (*DbClient, error) {
-	dbx, err := sql.Open(switchDriverName(driver), uri)
+	dbx, err := sql.Open(SwitchDriverName(driver), uri)
 	if err != nil {
 		return nil, fmt.Errorf("opening connection: %w", err)
 	}
 	err = dbx.Ping()
 	if err != nil {
 		if closeErr := dbx.Close(); closeErr != nil {
-			return nil, fmt.Errorf("failed to close with %s, after connection test failed: %w", closeErr, err)
+			return nil, fmt.Errorf("failed to close with %s, after connection test failed: %w", closeErr.Error(), err)
```

Comment:
Can we check and address the lint failure here.

Reply:
Yes. I have a solution ready.